dictionaries/software-terms/src/computing-acronyms.txt

```diff
@@ -75,3 +75,12 @@ XFS
 XML
 XSD
 ZIP
+UI # User Interface
+ADMIN
```

Comment:
Could you remind me how it works?

A word in uppercase without ~ it will accept the word but with what case?

Same question for a word with a ~

Reply:
@ccoVeille,

| caseSensitive | ADMIN | admin | Admin |
| --- | --- | --- | --- |
| `true` | found | not found | not found |
| `false` | found | found | found |

Unless told otherwise, the spell checker generates non-case sensitive versions of all words when it loads the dictionary.

Reply:
Thanks

To make sure ~ADMIN would have make it case insensitive, right?

Reply:
With People Names, we do not want the spell checker to generate non-case sensitive versions. So a directive is added to the header in the file:

https://github.com/streetsidesoftware/cspell-dicts/blob/0cbb972f4e2ebb49eb0c750482ea51cfc02c9c22/dictionaries/people-names/cspell-tools.config.yaml#L12-L13

https://github.com/streetsidesoftware/cspell-dicts/blob/0cbb972f4e2ebb49eb0c750482ea51cfc02c9c22/dictionaries/people-names/dict/people-names.txt#L3

Reply:
> To make sure ~ADMIN would have make it case insensitive, right?

Not really. `~ADMIN` would have just added it to the case insensitive list, but not actually change the case. In other words, `admin` would still not be found.

Here is an example with accents.

```
# cspell-dictionary: no-generate-alternatives
Café
~Cafe
```

| caseSensitive | Café | Cafe | cafe |
| --- | --- | --- | --- |
| `true` | found | not found | not found |
| `false` | found | found | no found |

Reply:
Thanks. But then I don't get something.

Either with what you explained or more likely what I understood.

ADMIN allows ADMIN only, but not admin.
~ADMIN allows ADMIN, but not admin.

I assume
admin allows admin, but not ADMIN
~admin allows ADMIN, admin, and Admin

But then why using ~ for other words in this file, like JSON, with uppercase.

Reply:
> - ADMIN allows ADMIN only, but not admin. 
> - ~ADMIN allows ADMIN, but not admin.

Yes.

> - admin allows admin, but not ADMIN 
> - ~admin allows ADMIN, admin, and Admin

No.
- `admin`: allows `admin`, `Admin`, `ADMIN`, and even `aDmIn` ([CSpell #2347](https://github.com/streetsidesoftware/cspell/issues/2347)).
- `~admin` is not needed if `admin` is already in the dictionary.

The `~` prefix is a way to add words to be used when case sensitivity is turned off. It does not change the letters.

Example:
```
English
~english
```

When case sensitive is enabled, then `english` would be an error.

Reply:
Ok but then, there is something I don't get

If we have English and ~english in a case sensitive dictionary.

Why would we need English ? When we have ~english ?

Is it because the English words might be in another dictionary, and adding english, would only allow English and english, but not enGlish?

I'm asking because I come back to the PR, in the file we have ~VGA and ~VHS, why would we need to have vga and vhs that are acronyms, with another syntax that uppercase?

I'm asking because the file is about acronyms, so mostly uppercase, except a few counter examples.

So you already had ~VGA, and now you add VGA.

It's the same for SVGA

But then the file contains  ~VPN, but not VPN

And another example, why is XML or ZIP without ~

Reply:
![Screenshot_20240812_213539_GitHub.jpg](https://github.com/user-attachments/assets/f0d469c9-a5a9-4f67-a214-3875383c4964)

Here, 👆

 but I have no idea why I see them in this PR.

I see nothing with a `~` in the history of this file in the last days

Reply:
@ccoVeille,

The `~` in the image is from the diff, it means approximate match. Notice that it is in line with the `+`, which means added.

Reply:
Thank you for the image. It helped a lot to understand the confusion.

Reply:
Apparently, it's something on the Android app of GitHub.

You can understand my confusion and why I was initially surprised to see things added without ~, while everything else in the file was with ~

Reply:
Makes sense. Thank you for raising the issue.